### DISCUSSION TITLE
Unable to upload content

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#757] [General Discussion]
Hello Team,

We are encountering an issue while uploading content. Below are the details:

**Environment Details:**
Knowledge Version: Release 5.2.0_RC5
API Endpoint: /action/content/v3/create
Issue Observed Since: Last week
Changes Made: No changes have been made recently to the system or configuration.

**Error Details:**
**Network Tab Response:**
`{
    "id": "api.content.create",
    "ver": "3.0",
    "ts": "2024-12-16T05:42:46ZZ",
    "params": {
        "resmsgid": "c6caf067-1c14-4fa9-8c51-cf13fec71c00",
        "msgid": null,
        "err": "ERR_EMPTY_EDGE_PROPERTY_LIST",
        "status": "failed",
        "errmsg": "The list to validate input is empty."
    },
    "responseCode": "CLIENT_ERROR",
    "result": {
        "messages": null
    }
}
`
**Log Details:**
`Default License is not available for channel: 
Exception occurred - class :org.sunbird.common.exception.ClientException with message :The list to validate input is empty.
2024-12-16 05:33:21,391 {"eid":"LOG","ets":1734327201391,"ver":"3.0","mid":"LP.1734327201391.580c7e08-960f-4a53-b2be-4794e4eb9540","actor":{"id":"org.sunbird.learning.platform","type":"System"},"context":{"channel":"in.ekstep","pdata":{"id":"production.production.portal","pid":"","ver":"1.0"},"env":"content","sid":"s0wuL3dZ_1E0I5ccr-eXLE6PetlpqKkl"},"edata":{"level":"INFO","type":"api_access","message":"","params":[{"duration":10},{"protocol":"http"},{"size":302},{"method":"POST"},{"uip":"10.1.10.187"},{"url":"/content/v3/create"},{"status":400}]},"syncts":1734327201391}
`

Thank you for your support!
Let me know if you'd like further refinements!

cc: @pallakartheekreddy

**Comment by pallakartheekreddy** [comment]
@siddhishinde0723 As error saying `Default License is not available for channel`. Please create a default license and update to channel.

**Reply by siddhishinde0723** [reply]
@pallakartheekreddy Is there any api for creating the default license?

Also we tried the same in dev env but there also i am geeting same error in log but I am able to upload the content.

**Reply by aimansharief** [reply]
@siddhishinde0723 Please use this API to create default license
`curl --location --globoff '{{host}}/api/license/v3/create' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer {{apikey}}' \
--data '{
    "request": {
        "license": {
            "name": "CC BY 4.0",
            "description": "This is the standard license of any content/youtube",
            "url": "https://creativecommons.org/licenses/by/4.0/legalcode"
        }
    }
}'`

**Reply by siddhishinde0723** [reply]
@aimansharief After running this api i am getting this error 
![image](https://github.com/user-attachments/assets/41cff179-b6dc-4a74-a980-2df9e1089f28)

**Reply by aimansharief** [reply]
@siddhishinde0723 If you are port forwarding content-service and triggering the API, use this curl - `curl --location 'localhost:9000/license/v3/create' \
--header 'Content-Type: application/json' \
--data '{
    "request": {
        "license": {
            "name": "CC BY 4.0",
            "description": "This is the standard license of any Youtube content",
            "url": "https://creativecommons.org/licenses/by/4.0/legalcode"
        }
    }
}'`

**Comment by siddhishinde0723** [comment]
@aimansharief Tried this also still facing the same issue

**Reply by aimansharief** [reply]
@siddhishinde0723 are you sure you port forwarded content service and then triggered this API?

**Reply by siddhishinde0723** [reply]
@aimansharief Yes.
Can we please connect?

**Reply by aimansharief** [reply]
@siddhishinde0723 can we connect tomorrow. I’m not available today

**Reply by siddhishinde0723** [reply]
@aimansharief The issue is solved.
After adding license in the redis this worked.

Thanks.

**Reply by aimansharief** [reply]
@siddhishinde0723 Did you manually add it in redis?

**Reply by aimansharief** [reply]
In the curl explicitly mention the http type. I'm sure this might work
`curl --location --request POST 'localhost:9000/license/v3/create' \
--header 'Content-Type: application/json' \
--data '{
    "request": {
        "license": {
            "name": "CC BY 4.0",
            "description": "This is the standard license of any Youtube content",
            "url": "https://creativecommons.org/licenses/by/4.0/legalcode"
        }
    }
}'`
